File: PTtools/R/PT/R/predictPT.R
.packageName <- 'PT'

predictPT <-function(delta, field, type="w"){

#PredPT <- data(PredPT);
if(type=="w"){

if(field=="R"){
x = PredPT$delta.Real
y = PredPT$eps.Real
# fit a spline to data of (delta,epsilon)
f<-splinefun(x=x,y=y) 
# predict at new delta
}


if(field=="C"){
x = PredPT$delta.Cplex
y = PredPT$eps.Cplex
# fit a spline to data of (delta,epsilon)
f<-splinefun(x=x,y=y) 
# predict at new delta
}

if(field=="Q" || field=="H2"){
x = PredPT$delta.Q
y = PredPT$eps.Q
# fit a spline to data of (delta,epsilon)
f<-splinefun(x=x,y=y) 

}

if(field=="O" || field=="H3"){
    x = PredPT$delta.O
    y = PredPT$eps.O
    # fit a spline to data of (delta,epsilon)
    f<-splinefun(x=x,y=y)
    # predict at new delta
}

if(field=="Bnd"){
f<- function(x){2*x-1}

}

if(field=="Pos"){
x = PredPT$delta.Pos
y = PredPT$eps.Pos
# fit a spline to data of (delta,epsilon)
f<-splinefun(x=x,y=y) 

}


}else if(type=="s"){

if(field=="R"){
    x = PredPT$delta.Real.S
    y = PredPT$eps.Real.S
    # fit a spline to data of (delta,epsilon)
    f<-splinefun(x=x,y=y)
    # predict at new delta
}


}

u = f(delta);

return(u * (u > 0) )

}

File: PTpredictions/MATLAB/predictPT.m
% 
% Predict PT 
% Author: Hatef Monajemi 
% Date: June 12 2012
% neps: number of epsilons 
% delta : scalar determining the shape factor (undersampling)
% flag: 'Real', 'Cplex', 'Pos' , 'Bnd'
% 
function eps_0 = predictPT(delta, flag)

PredPT = load('PredPT.mat');

switch flag 

case {'Real','R'}
eps_0 = interp1( PredPT.delta_Real, PredPT.eps_Real, delta,'PCHIP');

case {'Cplex','C'}
eps_0 = interp1( PredPT.delta_Cplex, PredPT.eps_Cplex, delta,'PCHIP');


case {'Pos', 'pos','R+'}
eps_0 = interp1( PredPT.delta_Pos, PredPT.eps_Pos, delta,'PCHIP');


case {'Bnd','bnd'}
eps_0 = interp1( PredPT.delta_Bnd, PredPT.eps_Bnd, delta,'PCHIP');


case 'Q'
eps_0 = interp1( PredPT.delta_Q, PredPT.eps_Q, delta,'PCHIP');
case 'O'
eps_0 = interp1( PredPT.delta_O, PredPT.eps_O, delta,'PCHIP');

otherwise
error('unknown flag')
end
